src/st/sel/3642C.c

```diff
@@ -32,7 +32,33 @@ void func_801B68E0(s32 arg0) {
     D_801BC390 = 1;
 }
 
-INCLUDE_ASM("asm/us/st/sel/nonmatchings/3642C", func_801B690C);
+void func_801B690C(u8 ySteps, Entity* self) {
+    s32 primIndex = g_Dialogue.D_801C24DA + 1;
+    Primitive* prim;
+    s32 i;
+
+    while (primIndex >= 5) {
+        primIndex -= 5;
+    }
+
+    if (self->step_s == 0) {
```

Comment:
Is self vs. g_CurrentEntity the only difference?

Reply:
Yes it is

Reply:
prim->v0 = ySteps + prim->v0;
also this, it doesn't match with prim->v0 += ySteps;

Reply:
https://decomp.me/scratch/Bdq1K